src/test/java/test/suites/github1850/DuplicateChildSuitesInitializationTest.java

```diff
@@ -0,0 +1,76 @@
+package test.suites.github1850;
+
+import org.testng.annotations.Test;
+import org.testng.xml.Parser;
+import org.testng.xml.XmlSuite;
+import test.SimpleBaseTest;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * This test checks that TestNG can handle duplicate child suites when we have the following set of files:
+ *
+ * - parent-suite-with-duplicate-child -> [child-suite-1, children/child-suite-3, child-suite1, children/child-suite-3]
+ * - children/child-suite-3 -> [../child-suite-2, child-suite-4, morechildren/child-suite-5]
+ *
+ * SHOULD return a XmlSuite object with following structure:
+ *
+ * parent-suite-with-duplicate-child
+ * 	├───child-suite-1
+ * 	├───child-suite-3
+ * 	│   ├───child-suite-2
+ * 	│   ├───child-suite-4
+ * 	│   └───child-suite-5
+ * 	├───child-suite-1(0)
+ * 	└───child-suite-3(0)
+ * 		├───child-suite-2(0)
+ * 		├───child-suite-4(0)
+ * 		└───child-suite-5(0)
+ *
+ * but NOT like:
+ *
+ * parent-suite-with-duplicate-child
+ * 	├───child-suite-1
+ * 	├───child-suite-3
+ * 	├───child-suite-1(0)
+ * 	└───child-suite-3(0)
+ * 		├───child-suite-2
+ * 		├───child-suite-4
+ * 		├─-─child-suite-5
+ * 		├───child-suite-2(0)
+ * 	    ├───child-suite-4(0)
+ * 	    └───child-suite-5(0)
+ *
+ * Check the <code>checksuitesinitialization</code> folder under test resources
+ *
+ * @author Chao Qin
+ */
+public class DuplicateChildSuitesInitializationTest extends SimpleBaseTest {
+
+
+
+ /**
+   * Parent suite contains duplicate child suites
+   */
+  @Test
+  public void checkDuplicateChildSuites() throws IOException {
```

Comment:
The assertions use index to run checks. It kind of makes it difficult for understanding and for maintenance as well. Was hoping that if we could basically also assert on the expected names for those indexes so that its easy to comprehend the test.